can/io/asc.py

```diff
@@ -155,7 +155,8 @@ def on_message_received(self, msg):
         if msg.is_extended_id:
             arb_id += "x"
 
-        channel = msg.channel if isinstance(msg.channel, int) else self.channel
+        # Many interfaces start channel numbering at 0 which is invalid
+        channel = msg.channel + 1 if isinstance(msg.channel, int) else self.channel
```

Comment:
I'll add channel support to ASCReader as well.

Only adding 1 if the channel is 0 is problematic though since then channel 0 and 1 would be the same.

Reply:
Uhm yeah. Right. But something like that ^^